File: homeworks/Karen_Kochinyan/ObjectCounter/src/objectcounter/ObjectCounter.java
/*
 TODO
 */
package objectcounter;

public class ObjectCounter {

    private static int objectsCount = 0;

    public static void main(String[] args) {
        ObjectCounter obj1 = new ObjectCounter();
        System.out.println(obj1.getObjectsCount());

        ObjectCounter obj2 = new ObjectCounter();
        ObjectCounter obj3 = new ObjectCounter();

        System.out.println(obj2.getObjectsCount());
        System.out.println(obj3.getObjectsCount());

        System.out.println(ObjectCounter.objectsCount);
    }

    public ObjectCounter() {
        ++objectsCount;
    }

    int getObjectsCount() {
        return ObjectCounter.objectsCount;
    }
}

File: homeworks/Karen_Kochinyan/lesson_1/object_counter.php
<?php

class A {

    public static $counter = 0;

    function __construct() {
        ++A::$counter;
    }

    function __destruct() {
        --A::$counter;
    }
    
    public function printObjectsCount() {
        print "<br/>Object count = " . A::$counter;
    }

}

$obj1 = new A();
$obj1->printObjectsCount();
$obj2 = new A();
$obj2->printObjectsCount();
$obj3 = new A();
$obj3->printObjectsCount();
$obj4 = new A();
$obj4->printObjectsCount();

unset($obj1);
$obj4->printObjectsCount();
unset($obj2);
$obj4->printObjectsCount();

?>